libs/providers/flagd/src/e2e/tests/in-process-provider.spec.ts

```diff
@@ -0,0 +1,88 @@
+import assert from 'assert';
+import { OpenFeature } from '@openfeature/server-sdk';
+import { FlagdProvider } from '../../lib/flagd-provider';
+import {
+  E2E_CLIENT_NAME,
+  FLAGD_NAME,
+  UNSTABLE_CLIENT_NAME,
+  UNAVAILABLE_CLIENT_NAME,
+  IMAGE_VERSION,
+} from '../constants';
+import { evaluation } from '../step-definitions/evaluation';
+import { GenericContainer, StartedTestContainer, TestContainer } from 'testcontainers';
+import { flagd } from '../step-definitions/flagd';
+import { flagdJsonEvaluator } from '../step-definitions/flagd-json-evaluator';
+import { flagdRecconnectUnstable } from '../step-definitions/flagd-reconnect.unstable';
+
+// register the flagd provider before the tests.
+async function setup() {
+  const containers: StartedTestContainer[] = [];
+
+  console.log('Setting flagd provider...');
+
+  const stable = await new GenericContainer(`ghcr.io/open-feature/sync-testbed:${IMAGE_VERSION}`)
+    .withExposedPorts(9090)
+    .start();
+  containers.push(stable);
+  OpenFeature.setProvider(
+    E2E_CLIENT_NAME,
+    new FlagdProvider({ resolverType: 'in-process', host: 'localhost', port: stable.getFirstMappedPort() }),
+  );
+
+  const unstable = await new GenericContainer(`ghcr.io/open-feature/sync-testbed-unstable:${IMAGE_VERSION}`)
+    .withExposedPorts(9090)
```

Comment:
```suggestion
    .withExposedPorts(9090)
```

Same as https://github.com/open-feature/js-sdk-contrib/pull/982/files#r1670560390, let's make this 8015 since that's the default and we have no clashing to worry about now.